### ISSUE TITLE
Bot will repeatedly set trunk as affected if we fix an issue via a pref-flip and represent that as `firefoxN: disabled`

### ISSUE THREAD

**Issue by dholbert** [mozilla/bugbot#2403]
If a bug is resolved as FIXED by turning off a pref, and we encode that in Bugzilla status flags by setting the status `firefoxN` to `disabled` (where N is current trunk), then we get into a state where bugbot will mark `firefoxN+1` to `affected` (and `firefoxN+2`, etc) when that version becomes trunk.

See for example https://bugzilla.mozilla.org/show_bug.cgi?id=1894614#c17 where this happened , and explanation in https://bugzilla.mozilla.org/show_bug.cgi?id=1894614#c19

This isn't a great situation to end up in.

We probably want to prevent this from happening, maybe by:
(1) Making the bot recognize this situation and not do anything.
or:
(2) having the bot detect this special situation and notify someone (assignee or triage owner) with a comment like the following (hastily written, please don't use this exact text :))
"Bug has been closed, but no version got a status `fixed`, so the assumption is that future Firefox versions should automatically be marked as affected. This is probably a mistake - should some Firefox version have a `fixed` status?"

**Comment by dholbert** [comment]
> (2) having the bot detect this special situation and notify someone (assignee or triage owner) with a comment like the following (hastily written, please don't use this exact text :))
> "Bug has been closed, but no version got a status `fixed`, so the assumption is that future Firefox versions should automatically be marked as affected. This is probably a mistake - should some Firefox version have a `fixed` status?"

Or maybe more generally:
(3) have the bot detect and nag about a broader situation that includes this scenario, i.e. bugs that affected some version that were closed as fixed but never had any version flagged as fixed. Wording for the nag might be along these lines:

"Bug is flagged as affecting some version, and was closed as FIXED, but no versions were set to status `fixed` (or `verified`). Could you set the status flag to `fixed` for the version where the fix landed, so we can reason about at-which-point future versions can also be implicitly considered fixed?"

**Comment by suhaibmujahid** [comment]
See also: https://github.com/mozilla/bugbot/issues/1535

**Comment by suhaibmujahid** [comment]
In such a scenario, the bug should probably show up for the release managers during their regression triage, and they can flag it as fixed when appropriate.

@rvandermeulen what do you think?

**Comment by dholbert** [comment]
Right, but that won't happen until after the bot has intervened to (incorrectly) mark e.g. `status-firefox128` as `affected`, which then requires folks to notice, figure-out-why-the-bot-did-that, and then clean up after it by resetting that flag and then resetting some earlier flag to the state that the bot was expecting.

That feels sub-optimal.

**Comment by dholbert** [comment]
(You're correct that it does show up on the release-manager regression-triage-list, and for now would be cleaned up by that team manually. The reason I noticed this issue in the first place because I'm part of the release manager regression-triage meeting for the 128 release cycle, and I happened to notice [the bug-in-question here](https://bugzilla.mozilla.org/show_bug.cgi?id=1894614#c17) in an early on-my-own skim of part of the bugs-to-be-triaged.  In my (limited) experience with the release-manager regression-triage-meetings so far, I'm noticing the meeting is quite time-constrained and doesn't always make it through the full bug list -- so if we can avoid making bugs unnecessarily show up in the lists there, it leaves more time for the folks in that meeting to get further in the list to other bugs that may actually need attention.)

**Comment by suhaibmujahid** [comment]
> That feels sub-optimal.

I totally agree. I want first to confirm with release managers that ignoring setting the flag for such cases would not unwanted impact any of their workflows/queries.

**Comment by rvandermeulen** [comment]
Discussed it with the team and TBH we don't have very strong opinions on this. We can see situations where a false positive could occur with or without this change. But overall, maybe it doesn't make sense to make `statusN+1` flag changes to bugs that are set to RESOLVED/VERIFIED? I can see the use case for `statusN-1` around uplifts, but what's the case higher release versions?

**Comment by dholbert** [comment]
FWIW:  I'm not necessarily suggesting that the bot not-set-the-flag in this situation.

I lean more towards just catching the situation earlier (around when the bug is closed) and poking the assignee to update the status flags at that point, if appropriate, per my suggested option (3) above ( https://github.com/mozilla/bugbot/issues/2403#issuecomment-2123029283 ), if the bot sees that the bug is closed and yet it can predict that it's going to mark future versions as affected.
